extensions/wikia/PhalanxII/PhalanxStatsSpecialController.class.php

```diff
@@ -42,87 +42,83 @@ public function index() {
 	}
 
 	private function blockStats($blockId) {
-		$this->wg->Out->setPageTitle( sprintf( "%s #%s", wfMsg('phalanx-stats-title'), $blockId ) );
-		$this->wg->Out->addBacklinkSubtitle( $this->title );
+		$out = $this->getOutput();
+		$out->setPageTitle( $this->msg( 'phalanx-stats-title' )->text() . "#$blockId" );
+		$out->addBacklinkSubtitle( $this->title );
 
 		$data = Phalanx::newFromId($blockId);
 
 		if ( !isset( $data["id"] ) ) {
-			$this->wg->Out->addWikiMsg( 'phalanx-stats-block-notfound', $blockId );
-			return;
+			$out->addWikiMsg( 'phalanx-stats-block-notfound', $blockId );
+			return false; // skip rendering
 		}
 
 		$data['author_id'] = User::newFromId($data['author_id'])->getName();
-		$data['timestamp'] = $this->wg->Lang->timeanddate( $data['timestamp'] );
+		$data['timestamp'] = $this->getLanguage()->timeanddate( $data['timestamp'] );
 
 		if ( $data['expire'] == null ) {
 			$data['expire'] = 'infinite';
 		} else {
-			$data['expire'] = $this->wg->Lang->timeanddate( $data['expire'] );
+			$data['expire'] = $this->getLanguage()->timeanddate( $data['expire'] );
 		}
 
 		$data['regex'] = $data['regex'] ? 'Yes' : 'No';
 		$data['case']  = $data['case']  ? 'Yes' : 'No';
 		$data['exact'] = $data['exact'] ? 'Yes' : 'No';
 		$data['lang'] = empty( $data['lang'] ) ? 'All' : $data['lang'];
 
-		if ( $data['type'] & Phalanx::TYPE_EMAIL && !$this->wg->User->isAllowed( 'phalanxemailblock' ) ) {
+		if ( $data['type'] & Phalanx::TYPE_EMAIL && !$this->getUser()->isAllowed( 'phalanxemailblock' ) ) {
 			/* hide email from non-privildged users */
-			$data['text'] = wfMsg( 'phalanx-email-filter-hidden' );
+			$data['text'] = $this->msg( 'phalanx-email-filter-hidden' )->escaped();
 		}
 
 		$data['type'] = implode( ', ', Phalanx::getTypeNames( $data['type'] ) );
 
-		/* stats table */
-		$headers = array(
-			wfMsg('phalanx-stats-table-id'),
-			wfMsg('phalanx-stats-table-user'),
-			wfMsg('phalanx-stats-table-type'),
-			wfMsg('phalanx-stats-table-create'),
-			wfMsg('phalanx-stats-table-expire'),
-			wfMsg('phalanx-stats-table-exact'),
-			wfMsg('phalanx-stats-table-regex'),
-			wfMsg('phalanx-stats-table-case'),
-			wfMsg('phalanx-stats-table-language'),
-		);
-
-		$tableAttribs = array(
-			'class' => 'wikitable',
-			'width' => '100%',
-		);
-
 		/* pull these out of the array, so they dont get used in the top rows */
 		$row = $data->toArray();
 		unset($row['text']);
 		unset($row['reason']);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "reason" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)